examples/api/lib/widgets/basic/custom_multi_child_layout.0.dart

```diff
@@ -0,0 +1,121 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for CustomMultiChildLayout
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(const ExampleApp());
+
+class ExampleApp extends StatelessWidget {
+  const ExampleApp({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Directionality(
+        // TRY THIS: Try changing the direction here and hot-reloading to
+        // see the layout change.
+        textDirection: TextDirection.ltr,
+        child: Scaffold(
+          body: ExampleWidget(),
+        ),
+      ),
+    );
+  }
+}
+
+/// Lays out the children in a cascade, where the top corner of the next child
+/// is a little above (`overlap`) the lower end corner of the previous child.
+///
+/// Will relayout if the text direction changes.
+class _CascadeLayoutDelegate extends MultiChildLayoutDelegate {
+  _CascadeLayoutDelegate({
+    required this.colors,
+    required this.overlap,
+    required this.textDirection,
+  });
+
+  final Map<String, Color> colors;
+  final double overlap;
+  final TextDirection textDirection;
+
+  // Perform layout will be called when re-layout is needed.
+  @override
+  void performLayout(Size size) {
+    final double columnWidth = size.width / colors.length;
+    Offset childPosition = Offset.zero;
+    switch (textDirection) {
+      case TextDirection.rtl:
+        childPosition += Offset(size.width, 0);
+        break;
+      case TextDirection.ltr:
+        break;
+    }
+    for (final String color in colors.keys) {
+      // layoutChild must be called exactly once for each child.
+      final Size currentSize = layoutChild(
+        color,
+        BoxConstraints(maxHeight: size.height, maxWidth: columnWidth),
+      );
+      // positionChild must be called to change the position of a child from
+      // what it was in the last layout. Each child starts at 0,0 initially.
+      switch (textDirection) {
+        case TextDirection.rtl:
+          positionChild(color, childPosition - Offset(currentSize.width, 0));
+          childPosition += Offset(-currentSize.width, currentSize.height - overlap);
+          break;
+        case TextDirection.ltr:
+          positionChild(color, childPosition);
+          childPosition += Offset(currentSize.width, currentSize.height - overlap);
+          break;
+      }
+    }
+  }
+
+  // shouldRelayout is called to see if the delegate has changed and requires
+  // a layout to occur. Should only return true if the delegate state itself
+  // changes: changes in the CustomMultiChildLayout attributes will
+  // automatically cause a relayout, like any other widget.
+  @override
+  bool shouldRelayout(_CascadeLayoutDelegate oldDelegate) =>
```

Comment:
IMHO the `=>` should be for one liners that return a value. This isn't a terribly long line.

```dart
bool shouldRelayout(_CascadeLayoutDelegate oldDelegate) => oldDelegate.textDirection != textDirection;
```